first_cake_spec.rb

```diff
@@ -0,0 +1,41 @@
+class FirstCake
+
+  attr_reader :name
+
+  def initialize(name)
+    @name = name
+  end
+
+  def self.all_da_cakes
+    ['New York Cheesecake', 'Birthday Cake', 'Black Forest Cake', 'Ice Cream Cake', 'Fruit Cake' ].map do |name|
+      new(name)
+    end
+  end
+end
+
+describe 'first cake' do
+  subject { FirstCake.all_da_cakes }
+
+  context '.all_da_cakes' do
+    specify 'returns all da cakes!' do
+
+      cake_array = []
+      subject.each do |cake|
+        cake_array << cake.name
+      end
```

Comment:
you could do subject.map &:name to achieve the same result (an array with the names of the cakes)